buffer/src/main/java/io/netty/buffer/api/internal/Statics.java

```diff
@@ -249,4 +262,72 @@ static void writeCharSequence(CharSequence source, Buffer destination, Charset c
         byte[] bytes = source.toString().getBytes(charset);
         destination.writeBytes(bytes);
     }
+
+    static boolean equals(Buffer bufferA, Buffer bufferB) {
+        if (bufferA == bufferB) {
+            return true;
+        }
+        final int aLen = bufferA.readableBytes();
```

Comment:
I think we miss null checks.  Especially if we use this in `Buffer.equals` as the parameter might be null